packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/InternalTypedRealm.kt

```diff
@@ -34,4 +38,67 @@ internal interface InternalTypedRealm : TypedRealm {
         val className = configuration.mediator.companionOf(clazz).`io_realm_kotlin_className`
         return ObjectQuery(realmReference, realmReference.schemaMetadata.getOrThrow(className).classKey, clazz, configuration.mediator, null, query, *args)
     }
+
+    private fun <T : BaseRealmObject> copyObjectFromRealm(obj: T, depth: Int, closeAfterCopy: Boolean, cache: ManagedToUnmanagedObjectCache): T {
+        // Be able to inject a cache here as well, so the Iterable case can share the cache
+        if (!obj.isManaged()) {
+            throw IllegalArgumentException("This object is unmanaged. Only managed objects can be copied: $obj.")
+        }
+        if (!obj.isValid()) {
+            throw IllegalArgumentException(
+                "Only valid objects can be copied from Realm. " +
+                    "This object was either deleted or the Realm has been closed, making this " +
+                    "object invalid: $obj."
+            )
+        }
+        if (depth < 0) {
+            throw IllegalArgumentException("Only a depth of 0 or more is allowed. Depth was: $depth.")
+        }
+        if (obj is RealmObjectInternal) {
+            val objectRef: RealmObjectReference<out BaseRealmObject> = obj.io_realm_kotlin_objectReference!!
+            val realmRef: RealmReference = objectRef.owner
+            val mediator: Mediator = realmRef.owner.configuration.mediator
+            return createDetachedCopy(mediator, obj, 0, depth, closeAfterCopy, cache)
+        } else {
+            throw IllegalStateException()
+        }
+    }
+
+    override fun <T : TypedRealmObject> copyFromRealm(obj: T, depth: Int, closeAfterCopy: Boolean): T {
+        return copyObjectFromRealm(obj, depth, closeAfterCopy, mutableMapOf())
+    }
+    override fun <T : TypedRealmObject> copyFromRealm(collection: Iterable<T>, depth: Int, closeAfterCopy: Boolean): List<T> {
+        val (valid: Boolean, nativePointer: NativePointer<*>?) = when (collection) {
+            is ManagedRealmList -> Pair(collection.isValid(), collection.nativePointer)
```

Comment:
you can fold 
```Kotlin
is ManagedRealmList, ManagedRealmSet
```
Since they return the same `Pair`

Reply:
This does not seem to work since the compiler cannot locate `isValid` in that case. I assume, because both are just normal instance methods on each class